cli/azd/pkg/containerapps/container_app.go

```diff
@@ -124,6 +130,28 @@ func (cas *containerAppService) DeployYaml(
 		return fmt.Errorf("decoding yaml: %w", err)
 	}
 
+	if shouldPersist := cas.alphaFeatureManager.IsEnabled(persistCustomDomainsFeature); shouldPersist {
+		aca, err := cas.getContainerApp(ctx, subscriptionId, resourceGroupName, appName)
+		if err == nil {
+			containerAppJson, err := json.Marshal(obj)
+			if err != nil {
+				panic("marshalling container app json failed")
+			}
+			var containerApp armappcontainers.ContainerApp
```

Comment:
I think this is not going to play nicely with the support we have for custom API versions - since the `armappcontainers.ContainerApp` model type here isn't going to match the shape of a custom version (in practice, I suspect this breaks the data protection stuff?). I think at a minimum you need to be partying `obj` in a "loosely typed" way.

Reply:
ah, I see.. yeah,, 

Updated to use `cofig` as a way to easily set the expected value :)